src/carplay.cs

```diff
@@ -0,0 +1,720 @@
+		CPTrip Trip { get; }
+
+		[Export ("updateTravelEstimates:forManeuver:")]
+		void UpdateTravelEstimates (CPTravelEstimates estimates, CPManeuver maneuver);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (CPTemplate))]
+	interface CPSearchTemplate {
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		ICPSearchTemplateDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+	}
+
+	interface ICPSearchTemplateDelegate { }
+	delegate void CPSearchTemplateDelegateUpdateHandler (CPListItem [] searchResults);
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface CPSearchTemplateDelegate {
+
+		[Abstract]
+		[Export ("searchTemplate:updatedSearchText:completionHandler:")]
+		void UpdatedSearchText (CPSearchTemplate searchTemplate, string searchText, CPSearchTemplateDelegateUpdateHandler completionHandler);
+
+		[Abstract]
+		[Export ("searchTemplate:selectedResult:completionHandler:")]
+		void SelectedResult (CPSearchTemplate searchTemplate, CPListItem item, Action completionHandler);
+
+		[Export ("searchTemplateSearchButtonPressed:")]
+		void SearchButtonPressed (CPSearchTemplate searchTemplate);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPSessionConfiguration {
+
+		[Export ("initWithDelegate:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ICPSessionConfigurationDelegate @delegate);
+
+		[Export ("limitedUserInterfaces")]
+		CPLimitableUserInterface LimitedUserInterfaces { get; }
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		ICPSessionConfigurationDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+	}
+
+	interface ICPSessionConfigurationDelegate { }
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[Protocol, Model ( AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface CPSessionConfigurationDelegate {
+
+		[Abstract]
+		[Export ("sessionConfiguration:limitedUserInterfacesChanged:")]
+		void LimitedUserInterfacesChanged (CPSessionConfiguration sessionConfiguration, CPLimitableUserInterface limitedUserInterfaces);
+	}
+
+	[Abstract]
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	interface CPTemplate : NSSecureCoding {
+
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPRouteChoice : NSCopying, NSSecureCoding {
+
+		[Export ("initWithSummaryVariants:additionalInformationVariants:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string [] summaryVariants, string [] additionalInformationVariants);
+
+		[Export ("summaryVariants", ArgumentSemantic.Copy)]
+		string [] SummaryVariants { get; }
+
+		[NullAllowed, Export ("additionalInformationVariants", ArgumentSemantic.Copy)]
+		string [] AdditionalInformationVariants { get; }
```

Comment:
This is `NullAllowed`, but the corresponding parameter in the constructor isn't? (and the same for `SummaryVariants`)

Reply:
yep, agreed, according to docs this should not be nil since it is a requirement in the ctor